src/cow-react/modules/limitOrders/containers/RateInput/index.tsx

```diff
@@ -56,14 +61,47 @@ export function RateInput() {
     return formatInputAmount(rate)
   }, [activeRate, areBothCurrencies, isInversed, isTypedValue, typedValue])
 
+  // TODO: refactor the logic and move it to a hook
```

Comment:
ideally we could have both a hook and a pure math function, for just the logic and price calculation.
The pure function is SDK material, the hook could remain in CoW Swap for now

Reply:
Done.
I'm gonna create a hook in the next PR, when I will be integrating it